internal/indexer/postgres/distributionsbylayer.go

```diff
@@ -33,47 +33,37 @@ func distributionsByLayer(ctx context.Context, db *sqlx.DB, hash claircore.Diges
 			   dist.pretty_name
 		FROM dist_scanartifact
 				 LEFT JOIN dist ON dist_scanartifact.dist_id = dist.id
-				 JOIN layer ON layer.hash = '%s'
+				 JOIN layer ON layer.hash = $1
 		WHERE dist_scanartifact.layer_id = layer.id
-		  AND dist_scanartifact.scanner_id IN (?);
+		  AND dist_scanartifact.scanner_id = ANY($2);
 		`
 	)
 
-	// TODO Use passed-in Context.
 	if len(scnrs) == 0 {
 		return []*claircore.Distribution{}, nil
 	}
+
 	// get scanner ids
-	scannerIDs := []int64{}
-	for _, scnr := range scnrs {
-		var scannerID int64
-		err := db.Get(&scannerID, selectScanner, scnr.Name(), scnr.Version(), scnr.Kind())
+	scannerIDs := make([]int64, len(scnrs))
+	for i, scnr := range scnrs {
+		err := s.pool.QueryRow(ctx, selectScanner, scnr.Name(), scnr.Version(), scnr.Kind()).
+			Scan(&scannerIDs[i])
 		if err != nil {
-			return nil, fmt.Errorf("store:distributionseByLayer failed to retrieve scanner ids for scnr %v: %v", scnr, err)
+			return nil, fmt.Errorf("store:distributionseByLayer failed to retrieve scanner ids for scanner %v: %v", scnr, err)
```

Comment:
house keeping: these error messages use a "id:" prefix which I don't think took off in the rest of the code base. Feel free to clean-up these error messages to make them uniform with the rest of the code.